### ISSUE TITLE
Keep being kicked out of Construct 3 editor (similar to #800?)

### ISSUE THREAD

**Issue by MeStrak** [Scirra/Construct-3-bugs#802]
## Problem description

I'm using Chrome (latest version). I keep being kicked out of the Construct 3 editor, back to the previous page I visited in the browser before loading the editor. This seems to happen when Construct 3 tries to display a popup (e.g. when creating an action, or building a project). When this happens I'm kicked out and lose all unsaved work.

If I cancel out of a Dialogue I have the 'Do you want to leave this page?' message that Chrome displays when leaving an active page - maybe this is related somehow.

## Attach a .c3p

https://www.dropbox.com/s/f657gzxmp6sv2na/Test%20%281%29.c3p?dl=0

## Steps to reproduce
Method 1
1. Open Construct 3 editor in Chrome
2. Go to project export
3. Select any item (tested with Cordova Android and HTML 5)
4. On minify script screen click next
5. Next screen loads very quickly (less than 0.5 seconds), then I am kicked out of the editor
6. Can also be reproduced from menus

Method 2
1. Add event
2. System
3. Compare variable
4. Kicked out of editor

## Observed result

Method 1: a popup loads very quickly then I'm kicked out and returned to page loaded before Construct editor
Method 2: I'm kicked out of the editor as soon as I double click 'Compare Variable'

## Expected result

I expected the next popup to load normally.

## Affected browsers

* Chrome: yes
* Firefox: no
* Edge: can't load construct 3
* Safari: note tested

## System details

*If you see a crash report dialog, please copy and paste it to where it says "PASTE HERE" below. Otherwise please go to **Menu** -> **About** -> **Platform information** and paste that information there instead.*

<details><summary>View details</summary>

Platform information
Browser: Chrome
Browser version: 61.0.3163.91
Browser engine: Blink
Browser architecture: 64-bit
Context: browser
Operating system: Windows
Operating system version: 10
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 4
Approx. device memory: (unavailable)
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.91 Safari/537.36
C3 release: r55

Local storage
Storage quota (approx): 30 gb
Storage usage (approx): 24 mb (0.1%)
Persistant storage: No

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Vendor: Google Inc.
Renderer: ANGLE (AMD Mobility Radeon HD 5000 Series Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_disjoint_timer_query_webgl2
EXT_texture_filter_anisotropic
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

Ogg Opus (audio/ogg; codecs=opus)
WebM Opus (audio/webm; codecs=opus)
Ogg Vorbis (audio/ogg; codecs=vorbis)
WebM Vorbis (audio/webm; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by AshleyScirra** [comment]
Does this still happen if you roll back to old builds? I'm having a hard time understanding how this is possible, especially since we haven't changed any of the related code in r55. Perhaps Chrome itself changed.

**Comment by MeStrak** [comment]
Hi Ashley,
I'm not sure, I only upgraded from my C2 membership a couple of days ago so this is the first build I've used. I'll look up how to rollback to an earlier build and try later. It seems quite intermittent - it started doing it last night, then stopped, and today it's back but even worse!

Are other users reporting the same? (I couldn't really tell if issue 800 is the same or not). It's possible it could be my PC, but I uninstalled and reinstalled Chrome and still had the same issue.

**Comment by MeStrak** [comment]
Ok, I've had the following error message a few times too, but hadn't taken a screenshot before and I forgot to mention it in the initial post. I've no idea if it's related but maybe it can help.

![image](https://user-images.githubusercontent.com/31989238/30498127-ce847432-9a55-11e7-9154-d7e11a29c33c.png)

I've disabled all installed browser addons but I still have the same issue.

I also found another way to reproduce it:
1. Start opening a project (mine are on Google Drive)
2. Hit Cancel before the load is complete
3. 'dialog not open' message displayed
4. Try to open the project again
5. Progress bar completes, but C3 doesn't open, I'm kicked back to the browser start page

I'm sorry but I don't know how to rollback to a previous build, can you send me a link to some instructions?

**Comment by MeStrak** [comment]
I also just downloaded Opera as a workaround and it's doing exactly the same thing.

**Comment by ospishchev** [comment]
I have the same problem as MeStrak! 

> I keep being kicked out of the Construct 3 editor, back to the previous page...

Everything is absolutely as he said. I'm using the latest version of Crome on macOS Sierra. Most likely it all started after the upgrade to r54 or r55, but not earlier. The same problem on OpenSUSE leap 42.3 with the latest Chrome. I also kicked out when I add a new action or event. And a pop-up window with a suggestion to "leave the page/stay" always appears. And sometimes I can not use some letters (x, r, a, ...), they just are not typed in variable names, but letters (X, R, A ....) are typed! I deleted all add-ons of Chrome; I re-installed chrome, I re-installed the operating system - nothing helps!
